cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake

```
@@ -6,70 +6,90 @@ macro(SetupShibokenAndPyside)
+            set(PySide_INCLUDE_DIR ${PySide${PYSIDE_MAJOR_VERSION}_INCLUDE_DIRS})
+            set(PySide_LIBRARY ${PySide${PYSIDE_MAJOR_VERSION}_LIBRARIES})
+            set(PySide_FOUND TRUE)
+        endif()
+    endif()
 
-    find_package(PySide2Tools QUIET) #REQUIRED # PySide2 utilities (pyside2-uic & pyside2-rcc)
-    if(NOT PYSIDE2_TOOLS_FOUND)
+    find_package(PySide${PYSIDE_MAJOR_VERSION}Tools QUIET) # PySide utilities (uic & rcc executables)
+    if(NOT PYSIDE_TOOLS_FOUND)
         message("=======================\n"
-                "PySide2Tools not found.\n"
+                "PySide${PYSIDE_MAJOR_VERSION}Tools not found.\n"
                 "=======================\n")
-    endif(NOT PYSIDE2_TOOLS_FOUND)
+    endif()
 
     file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/Ext/PySide)
     file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/__init__.py "# PySide wrapper\n"
-                                  "from PySide2 import __version__\n"
-                                  "from PySide2 import __version_info__\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtCore.py "from PySide2.QtCore import *\n\n"
+                                "from PySide${PYSIDE_MAJOR_VERSION} import __version__\n"
+                                "from PySide${PYSIDE_MAJOR_VERSION} import __version_info__\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtCore.py "from PySide${PYSIDE_MAJOR_VERSION}.QtCore import *\n\n"
                                 "#QCoreApplication.CodecForTr=0\n"
                                 "#QCoreApplication.UnicodeUTF8=1\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtGui.py  "from PySide2.QtGui import *\n"
-                                "from PySide2.QtWidgets import *\n"
-                                "QHeaderView.setResizeMode = QHeaderView.setSectionResizeMode\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtSvg.py  "from PySide2.QtSvg import *\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtUiTools.py  "from PySide2.QtUiTools import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtNetwork.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtNetwork import *\n")
+    if(BUILD_GUI)
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtGui.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtGui import *\n"
+                                    "from PySide${PYSIDE_MAJOR_VERSION}.QtWidgets import *\n"
+                                    "QHeaderView.setResizeMode = QHeaderView.setSectionResizeMode\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtSvg.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtSvg import *\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtUiTools.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtUiTools import *\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWidgets.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtWidgets import *\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWebEngineWidgets.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtWebEngineWidgets import *\n")
```

```suggestion
        if(BUILD_WEB)
          file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWebEngineWidgets.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtWebEngineWidgets import *\n")
        endif()
```

So it's like in https://github.com/FreeCAD/FreeCAD/pull/7647/files#diff-3cb8f98be3171f8d8c63baa371de5d5b5f511e843b8bdc201b254dc0cd215c62R12-R13

---

It's actually not correct, though. The Web workbench doesn't use the PySide QtWebEngineWidgets import at all, it's the Addon Manager that uses it (or will, once it exists 😄). Actually, if you disable Web, because we then disable QtWebEngineWidgets, several things break, Start and Addon Manager among them. That's probably something we should clean up later on.

---

I know about Start needing QtWebEngine. It's the reason why I removed the option to enable/disable the Web module on my FreeCAD package. It doesn't really make sense to omit the Start workbench on a BUILD_GUI build.

Do I get you right here? So the Web workbench uses QtWebEngine (C++), while Start and AddonManager use the PySide bindings of QtWebEngine?

---

There are several forums denizens who don't like, and don't use, Start -- you can configure FreeCAD to show a different Workbench on startup, and apparently they do (I personally do use Start, but it's not unreasonable to compile FreeCAD without it.

>  So the Web workbench uses QtWebEngine (C++), while Start and AddonManager use the PySide bindings of QtWebEngine?

Not quite -- Start uses the Web workbench, so by extension uses the C++ calls. If you disable Web, Start is also disabled automatically. The Addon Manager uses the PySide import, but that only actually works if FreeCAD includes QtWebEngineWidgets in its compilation -- which it only does if Web is enabled. That said, the Addon Manager can run without it, you just don't get the nice rendered version of the README files.